### PR TITLE
Add Android 5+ support

### PR DESCRIPTION
**Pull request by larryng**
Merges in https://github.com/libusb/libusb/pull/242.  Tested locally in https://github.com/ButterflyNetwork/software/pull/6800.

Please take a close look since there were some merge conflicts and I hardly know what I'm doing in both libusb and C++.

### CODE REVIEW

**Comment by jgavris** [comment]
Android 5+ support?

**Comment by jgavris** [comment]
We know this works on 8 and 9?

**Comment by larryng** [comment]
"5+" is probably better.  The key difference is a permission change for the stricter starting on 5.  I've been testing 8 and it works.  Probably works on 9, too, but we can find that out for sure.

File: libusb/core.c [hunk 1]
```diff
@@ -1211,6 +1211,74 @@ int usbi_clear_event(struct libusb_context *ctx)
...
+	size_t priv_size = usbi_backend.device_handle_priv_size;
+	int r;
+	usbi_dbg("wrap %d", fd);
+
+	USBI_GET_CONTEXT(ctx);
+
+	if (!usbi_backend.wrap_fd)
+		return LIBUSB_ERROR_NOT_SUPPORTED;
+
+	_dev_handle = malloc(sizeof(*_dev_handle) + priv_size);
```
Comment (jgavris) [inline libusb/core.c:1251]: This is freed in `libusb_close` / `do_close`?

**Comment by kersson** [comment]
What were the merge conflicts?

**Comment by larryng** [comment]
Merge conflicts were around [here](https://github.com/ButterflyNetwork/libusb/pull/14/files#diff-80293e4e004c60e0285efa11de2f6b33R1388).  I think `op_open_fd` is our own homegrown function, and this patch adds `op_wrap_fd` along with some refactoring around `op_open` including extracting out `initialize_handle`.  I _think_ resolving the conflicts were mostly straightforward reshuffling of some lines, but I could be missing something.

**Comment by kersson** [comment]
IIRC, I added `op_open_fd` stuff back in the day when I added Android support myself. But given that you're ripping all that out, maybe we should just rebase those commits out?

**Comment by larryng** [comment]
Oh, you said rebase.  Let me see.

**Comment by jgavris** [comment]
You can’t rebase...revert those commits in a PR before applying the merge.

**Comment by larryng** [comment]
Yep

**Comment by larryng** [comment]
Updated.

**Review by jgavris — APPROVED** [review]
LGTM

**Comment by kersson** [comment]
So there were no conflicts after reverting?

**Comment by larryng** [comment]
No conflicts

**Comment by larryng** [comment]
I mean, technically no merge conflicts, but a tiny fix I had to make was changing a few `usbi_backend->wrap_fd` to `usbi_backend.wrap_fd` but that's it.

**Review by kersson — APPROVED** [review]

**Pull request merged**